src/models/SnackbarStore.ts

```diff
@@ -0,0 +1,70 @@
+import { SnapshotOrInstance, types, cast, Instance, flow, getRoot, SnapshotIn } from 'mobx-state-tree'
+import { autorun } from 'mobx'
+import { createId } from '@/utils/createId'
+
+export type SnackbarIconType = 'success' | 'error' | 'info' | 'warning'
+
+const SNACKBARS_LIMIT = 3
+
+export const Snackbar = types
+  .model('Snackbar', {
+    id: types.optional(types.identifier, () => createId()),
+    title: types.string,
+    description: types.maybe(types.string),
+    timeout: types.maybe(types.number),
+    variant: types.maybe(types.enumeration(['primary', 'secondary'])),
+    iconType: types.maybe(types.enumeration(['success', 'error', 'info', 'warning'])),
+    actionText: types.maybe(types.string),
+  })
+  .volatile((self) => ({
+    // eslint-disable-next-line @typescript-eslint/no-empty-function
+    onActionClick() {},
+  }))
+  .actions((self) => ({
+    setOnActionClick(onActionClick: () => void) {
+      self.onActionClick = onActionClick
+      return self
+    },
+    updateSnackbar(snackbar: ISnackbar) {
+      self = { ...self, ...snackbar }
+    },
+  }))
+
+export const SnackbarStore = types
+  .model('SnackbarStore', {
+    snackbars: types.array(Snackbar),
+  })
+  .actions((self) => ({
+    afterAttach() {
```

Comment:
This part is a little confusing for me. Why do we need to do this? And why we use `autorun` and pass empty function below?

Reply:
So, afterAttach is a lifecycle hook that's called when the model is attached to the tree and autorun is like an useEffect that handles dependencies automatically so it runs once and whenever something inside changes it will rerun again

Oh, removed the empty autorun not sure how it ended up in there.